src/algebra/covariant_and_contravariant.lean

```diff
@@ -196,6 +197,41 @@ end is_trans
 
 end contravariant
 
+section monotone
+
+variables {α : Type*} {M N μ} [preorder α] [preorder N]
+variable {f : N → α}
+
+/-- The partial application of a constant to a covariant operator is monotone. -/
+lemma covariant.monotone_of_const [covariant_class M N μ (≤)] (m : M) : monotone (μ m) :=
+  λ a b ha, covariant_class.elim m ha
+
+/-- A monotone function remains monotone when composed with the partial application
+of a covariant operator. E.g., `∀ (m : ℕ), monotone f → monotone (λ n, f (m + n))`. -/
+lemma monotone.covariant_of_const [covariant_class M N μ (≤)] (hf : monotone f) (m : M) :
+  monotone (f ∘ μ m) :=
+hf.comp $ covariant.monotone_of_const m
+
+/-- Same as `monotone.covariant_of_const`, but with the constant on the other side of
+the operator.  E.g., `∀ (m : ℕ), monotone f → monotone (λ n, f (n + m))`. -/
+lemma monotone.covariant_of_const' {μ : N → N → N} [covariant_class N N (swap μ) (≤)]
+  (hf : monotone f) (m : N) :
+  monotone (f ∘ (swap μ) m) :=
```

Comment:
```suggestion
  monotone (λ n, f (μ n m)) :=
```
Is there a particular reason to write this in pointfree style? We generally prefer to avoid it if possible because `∘` and `swap` aren't always unfolded when we want them to be.

Reply:
IIRC, a lot of lemmas in `monotone.lean` use composition and [Yuri's comment](https://github.com/leanprover-community/mathlib/pull/11689#discussion_r795897682) used partial application, so I thought it was more idiomatic. I'm happy to change this though. Lemme check that everything still works.

Reply:
@urkud, thoughts? You are closer to this material than I.